node/src/main/scala/coop/rchain/node/web/ReportingRoutes.scala

```diff
@@ -1,111 +1,64 @@
+import coop.rchain.metrics.{Metrics, MetricsSemaphore, Span}
+import org.http4s.HttpRoutes
+import coop.rchain.shared.Log
 import org.http4s.circe.jsonEncoderOf
 
-object ReportingRoutes {
+import scala.collection.concurrent.TrieMap
 
-  val prettyPrinter = RhoPrinter()
-  val transformer =
-    new ReportingEventStringTransformer[Par, BindPattern, ListParWithRandom, TaggedContinuation](
-      serializeC = channel => prettyPrinter.buildString(channel),
-      serializeP = p => p.patterns.map(prettyPrinter.buildString).mkString("[", ";", "]"),
-      serializeA = data => data.pars.map(prettyPrinter.buildString).mkString("[", ";", "]"),
-      serializeK = k =>
-        k.taggedCont match {
-          case ParBody(value)      => prettyPrinter.buildString(value.body)
-          case ScalaBodyRef(value) => s"ScalaBodyRef($value)"
-          case Empty               => "empty"
-        }
-    )
+object ReportingRoutes {
 
-  final case class DeployTrace(deployHash: String, source: String, events: List[List[RhoEvent]])
-  sealed trait Report
+  sealed trait ReportResponse
   final case class BlockTracesReport(
-      hash: String,
-      traces: List[DeployTrace]
-  ) extends Report
-  final case class BlockNotFound(hash: String)    extends Report
-  final case class BlockReplayError(hash: String) extends Report
-  final case class BlockReplayFailed(hash: String, msg: String, deployId: Option[String])
-      extends Report
+      report: BlockEventInfo
+  ) extends ReportResponse
+  final case class BlockReportError(hash: String, errorMessage: String) extends ReportResponse
 
-  def transformDeploy[C, P, A, K](transformer: ReportingTransformer[C, P, A, K, RhoEvent])(
-      ipd: ProcessedDeploy,
-      events: Seq[Seq[ReportingEvent]]
-  ): DeployTrace =
-    DeployTrace(
-      ipd.deploy.sig.base16String,
-      ipd.deploy.data.term,
-      events.map(a => a.map(transformer.transformEvent).toList).toList
-    )
+  type ReportingHTTPRoutes[F[_]] = HttpRoutes[F]
 
   def transforResult[F[_]: Sync](
-      hash: BlockHash,
-      state: F[
-        Either[ReportError, List[(ProcessedDeploy, Seq[Seq[ReportingEvent]])]]
-      ]
-  ): F[Report] =
+      hash: String,
+      state: F[ApiErr[BlockEventInfo]]
+  ): F[ReportResponse] =
     state.map(
       s =>
         s match {
-          case Left(ReportBlockNotFound(hash)) => BlockNotFound(hash.base16String)
-          case Left(ReportReplayError(error)) =>
-            BlockReplayFailed(
-              hash.base16String,
-              error.toString,
-              None
-            )
-          case Right(deploys) =>
-            val t = transformDeploy(transformer)(_, _)
-            BlockTracesReport(
-              hash.base16String,
-              deploys.map(Function.tupled(t))
-            )
-          case _ => BlockReplayError(hash.base16String)
+          case Left(error) => BlockReportError(hash, error)
+          case Right(b)    => BlockTracesReport(b)
```

Comment:
This can be simplified to pattern match with anonymous function.
```scala
state.map {
  case Left(error) => BlockReportError(hash, error)
  case Right(b) => BlockTracesReport(b)
}
```